src/qml/QFieldGuide.qml

```diff
@@ -0,0 +1,327 @@
+    property var step: steps[index]
+    property var target: {
+      if (steps[index]) {
+        return steps[index].target();
+      }
+      return undefined;
+    }
+    property int parentHeight: {
+      if (guide.parent) {
+        return guide.parent.height;
+      }
+      return guide.height;
+    }
+    property int parentWidth: {
+      if (guide.parent) {
+        return guide.parent.width;
+      }
+      return guide.width;
+    }
+  }
+
+  Connections {
+    target: internalObject.window
+    function onWidthChanged() {
+      canvas.requestPaint();
+      delayedPainter.restart();
+    }
+    function onHeightChanged() {
+      canvas.requestPaint();
+      delayedPainter.restart();
+    }
+  }
+
+  // not to draw for every frame of resizing
+  Timer {
+    id: delayedPainter
+    interval: 200
+    onTriggered: {
+      canvas.requestPaint();
+    }
+  }
+
+  Canvas {
+    id: canvas
+    anchors.fill: parent
+    onPaint: {
+      internalObject.pos = internalObject.target.mapToGlobal(0, 0);
+      internalObject.pos = Qt.point(internalObject.pos.x - internalObject.window.x, internalObject.pos.y - internalObject.window.y);
+      var ctx = canvas.getContext("2d");
+      ctx.clearRect(0, 0, canvasSize.width, canvasSize.height);
+      ctx.save();
+      ctx.fillStyle = "#88000000";
+      ctx.fillRect(0, 0, canvasSize.width, canvasSize.height);
+      ctx.globalCompositeOperation = 'destination-out';
+      ctx.fillStyle = 'black';
+      var rect = Qt.rect(internalObject.pos.x - guide.targetMargins, internalObject.pos.y - guide.targetMargins, internalObject.target.width + guide.targetMargins * 2, internalObject.target.height + guide.targetMargins * 2);
+      drawRoundedRect(rect, 8, ctx);
+      ctx.restore();
+    }
+
+    function drawRoundedRect(rect, r, ctx) {
+      ctx.beginPath();
+      ctx.moveTo(rect.x + r, rect.y);
+      ctx.lineTo(rect.x + rect.width - r, rect.y);
+      ctx.arcTo(rect.x + rect.width, rect.y, rect.x + rect.width, rect.y + r, r);
+      ctx.lineTo(rect.x + rect.width, rect.y + rect.height - r);
+      ctx.arcTo(rect.x + rect.width, rect.y + rect.height, rect.x + rect.width - r, rect.y + rect.height, r);
+      ctx.lineTo(rect.x + r, rect.y + rect.height);
+      ctx.arcTo(rect.x, rect.y + rect.height, rect.x, rect.y + rect.height - r, r);
+      ctx.lineTo(rect.x, rect.y + r);
+      ctx.arcTo(rect.x, rect.y, rect.x + r, rect.y, r);
+      ctx.closePath();
+      ctx.fill();
+    }
+  }
+
+  Rectangle {
+    id: hintPannel
+    width: Math.max(250, mainWindow.width / 2)
+    height: 88 + description.height + (animatedHint.visible ? animatedHint.height + 8 : 20)
+    radius: 4
+    clip: true
+
+    Behavior on x  {
+      enabled: enablePannelAnimation
+
+      NumberAnimation {
+        duration: 150
+      }
+    }
+    Behavior on y  {
+      enabled: enablePannelAnimation
+
+      NumberAnimation {
+        duration: 150
+      }
+    }
+
+    property color color: Theme.mainBackgroundColor
```

Comment:
Again, let's make sure properties are sitting at the beginning of items (below ID).